### ISSUE TITLE
Phase D: VTL vs NumPy/PyTorch Benchmark Suite

### ISSUE THREAD

**Issue by ulises-jeremias** [vlang/vtl#88]
## Problem

VTL needs quantitative performance comparison against NumPy (CPU baseline) and PyTorch (GPU baseline) to track optimization progress.

## Proposed Solution

Create `vtl/benchmarks/vs_numpy/` with CI-integrated benchmarks:

### `matmul.v` — Matmul benchmark

```v
import vtl
import vtl.la
import benchmark

fn benchmark_matmul(n int, iterations int) {
    a := vtl.rand[f64]([n, n])
    b := vtl.rand[f64]([n, n])
    mut bm := benchmark.new_benchmark()
    // warmup + timed runs
    c := vtl.la.matmul(a, b)!
}
```

### CI Integration (GH Actions)

- Run on every PR
- Post comment: VTL time, NumPy time, speedup ratio
- Fail if VTL >3x slower than NumPy (with justification allowed)

## Priority Benchmarks

1. **Matmul** — `f64[1024, 1024]` CPU and CUDA
2. **Conv2D** — `f64[64, 3, 32, 32]` kernel `f64[3, 3, 3, 64]`
3. **Autograd backprop** — 3-layer MLP, 256 hidden
4. **Training step** — forward + loss + backprop + optimizer update

## NumPy Reference

```python
# Matmul
python -c "import numpy as np; a=np.random.rand(1024,1024); b=np.random.rand(1024,1024); print(%timeit np.dot(a,b))"

# Conv2D
python -c "import numpy as np; x=np.random.rand(64,3,32,32); w=np.random.rand(64,3,3,3); print(%timeit np.einsum('bchw,cohw->bohw', x, w))"
```

## Acceptance Criteria

- [ ] `benchmarks/vs_numpy/matmul.v` runs and reports GFLOPS
- [ ] GH Actions workflow triggered on PR
- [ ] Benchmark results posted as PR comment

## Dependencies

- Depends on: VTL matmul working (done)
- Blocks: Performance optimization tracking

## Status

Open — not yet assigned
